src/lib/utils/log_manager.cpp

```diff
@@ -0,0 +1,26 @@
+#include "log_manager.hpp"
+
+#include "constant_mappings.hpp"
+#include "utils/settings/log_level_setting.hpp"
+
+namespace opossum {
+
+LogManager::LogManager(SettingsManager& settings_manager, LogLevel log_level) : _log_level(log_level) {
+  if (!settings_manager.has_setting(LOG_LEVEL_SETTING_NAME)) {
+    auto log_level_setting = std::make_shared<LogLevelSetting>(LOG_LEVEL_SETTING_NAME);
+    log_level_setting->register_at(settings_manager);
+  }
+}
+
+void LogManager::add_message(const std::string& reporter, const std::string& message, const LogLevel log_level) {
+  if (log_level < _log_level) return;
```

Comment:
I think my idea of the LogLevel was a different one: Instead of checking the level during adding a message to the manager, we only return the messages with a certain level. That way all messages would be logged but only certain ones retrieved.

Reply:
Would you prefer filtering via select (and thus, maybe need to address multiple values in the where clause) or keeping the minimal level as a setting?

Reply:
Maybe we should have another look at what we want to accomplish with the log levels. Looking at it again, I don't see a purpose for the `error` level and I question the `warning` level. By design, a Hyrise process does not have any loggable errors, because each detected error leads to std::terminate.

Reply:
I envisioned it to be filtered via select otherwise I imagine cases where the minimal level is frequently changed back and forth.

Reply:
> By design, a Hyrise process does not have any loggable errors

Right, there are nor errors in Hyrise.

Reply:
Use case for log levels: Our current (BP) plugins run periodically and log that they are running, initialized and stopped. In addition, they add another message if they actually adjust the configuration. Plugins produce a lot of log messages, some of them are more important than others.

Reply:
Yes, I just used the "common" levels used anywhere and expected to have a discussion about them. And due _to let if fail_ I guess we can keep Debug and Info (and maybe warning for some cases).

If we only use select to filter, also the setting can be removed and the SettingsManager can be removed from LogManager's constructor. I think this would be okay when using not more than three log levels (so filtering via select gets a little more handy).

Reply:
Note that filtering by log level is somewhat painful right now. We would need a numerical representation of the log level, too.

Reply:
When doing it by select statements, yes, one needs to type every level in the where clause, but this could be okay with just a few levels. This was the reason to introduce a setting with a minimal level, as we can compare log levels internally via their enum values.